### PR TITLE
Delete unused compression formats from .import files when exporting

### LINKED ISSUE #74451: Godot attempting to load BPTC textures on Android
### Godot version

Godot 4.0 Stable

### System information

Windows 11, Vulkan Mobile, NVidia RTX 3070 TI - targeting Quest 2

### Issue description

When running the Godot-XR-Tools demo project natively on the Quest, the main scene fails to load because the resource loader tries to load the BPTC version of the texture, even though the ASTC version was bundled into the APK.

Unable to open file: res://.godot/imported/SkyOnlyHDRI023_2K-TONEMAPPED.jpg-bcb5464e5ed42ee2463b71652f907f38.bptc.ctex.

### Steps to reproduce

Build the latest godot-xr-tools demo project (https://github.com/GodotVR/godot-xr-tools) targeting the Quest.

The project has both S3TC/BPTC and ETC2/ASTC texture compression formats enabled.

The .import files mention both the S3TC/BPTC and ETC2/ASTC formats; however when exporting for Quest 2 (an Android target) Godot automatically enables the astc and etc2 features and bundles only the ASTC and ETC2 textures into the APK.

At startup the resource importer reads the .import files, sees S3TC/BPTC as the first enabled texture, sees that S3TC/BPTC is supported on the platform, and picks that format - totally ignoring what was actually bundled into the installer.

### Minimal reproduction project

Build the latest godot-xr-tools demo project (https://github.com/GodotVR/godot-xr-tools) targeting the Quest.

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/74451

This strips the unused compression formats out of the exported copy of the .import file. This way the engine will only attempt to load textures that have been exported. 

In #74451 I said that part of the change we would like to make is to unify the compression export options with the import options. However, I think that is better left to another PR as it will change behaviour and this is purely a bugfix.

@Malcolmnixon Could you test on the Quest? I have tested on a device that supports all 4 formats and can confirm that this fixes the issue on it. But I have not tested on Android.

### CODE REVIEW

File: editor/export/editor_export_platform.cpp [hunk 1]
```diff
@@ -1138,9 +1141,17 @@ Error EditorExportPlatform::export_project_files(const Ref<EditorExportPreset> &
 					return err;
 				}
 
-				//also save the .import file
-				Vector<uint8_t> array = FileAccess::get_file_as_bytes(path + ".import");
-				err = p_func(p_udata, path + ".import", array, idx, total, enc_in_filters, enc_ex_filters, key);
+				// erase useless sections
```
Comment (akien-mga) [inline editor/export/editor_export_platform.cpp:1144]:
```suggestion
				// Erase useless sections.
```
While at it, you could fix the style and casing of `// Erase all PAths` just before the chunk of code you're modifying.

Comment (clayjohn) [reply editor/export/editor_export_platform.cpp]: Fixed. And I also fixed up the nearby comments too.

**Comment by Malcolmnixon** [comment]
I tested this on the Quest and it seems to work perfectly. Specifically:
 * The .import file is highly stripped down and contains only the path to the exported ASTC texture
 * It runs on the Quest 2 headset

For reference this is a stripped down version of one of the .import files:
```
[remap]

importer="texture"
type="CompressedTexture2D"
uid="uid://dqoe1c5yn5dnv"
path.astc="res://.godot/imported/SkyOnlyHDRI023_2K-TONEMAPPED.jpg-bcb5464e5ed42ee2463b71652f907f38.astc.ctex"
metadata={
"imported_formats": ["s3tc_bptc", "etc2_astc"],
"vram_texture": true
}
```

It might theoretically be possible to strip the "s3tc_bptc" entry from "imported_formats" - the host doesn't need to know which formats the development PC imported - just the one available in the imports. However I don't think it's really a big issue as it doesn't cause an issue - it's just an information leak.

**Comment by Malcolmnixon** [comment]
Was there any additional testing needed for this to go in?

It would be nice to try and target this fix for the 4.0.1 release if possible. Currently the VR developers are having to erroneously add s3tc and bptc support to their android exports in order to prevent loading issues.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.2.

**Pull request merged**